dashboard/views.py

```diff
@@ -134,6 +137,90 @@ def get_course_info(request, course_id=0):
+
+    course_id = canvas_id_to_incremented_id(course_id)
+    current_user = request.user.get_username()
+
+    bad_json_response = HttpResponseBadRequest(JsonResponse({'error': 'Request JSON malformed.'}))
+
+    try:
+        request_data: dict = json.loads(request.body.decode('utf-8'))
+    except JSONDecodeError:
+        return bad_json_response
+
+    schema = {'$schema': 'http://json-schema.org/draft-07/schema',
+              'type': 'object',
+              'additionalProperties': False,
+              'properties': {
+                  'ap': {
+                      'type': 'object',
+                      'required': ['enabled'],
+                      'additionalProperties': False,
+                      'properties': {'enabled': {'type': 'boolean'}}},
+                  'apv1': {
+                      'type': 'object',
+                      'required': ['enabled'],
+                      'additionalProperties': False,
+                      'properties': {'enabled': {'type': 'boolean'}}},
+                  'gd': {
+                      'type': 'object',
+                      'required': ['enabled'],
+                      'additionalProperties': False,
+                      'properties': {'enabled': {'type': 'boolean'},
+                                     'show_grade_counts': {'type': 'boolean'}}},
+                  'ra': {
+                      'type': 'object',
+                      'required': ['enabled'],
+                      'additionalProperties': False,
+                      'properties': {'enabled': {'type': 'boolean'}}}},
+              'minProperties': 1}
+
+    try:
+        jsonschema.validate(request_data, schema)
+    except jsonschema.ValidationError:
+        return bad_json_response
+
+    # to translate short names returned by model back to original column names
+    view_column_names: dict = {
+        'ap': CourseViewOption.show_assignment_planning.field_name,
+        'apv1': CourseViewOption.show_assignment_planning_v1.field_name,
+        'gd': CourseViewOption.show_grade_distribution.field_name,
+        'ra': CourseViewOption.show_resources_accessed.field_name
+    }
+
+    view_settings: dict
+    view_data: dict = {}
+    success: bool = True  # always look on the bright side of life
+
+    try:
+        for (view_key, view_settings) in request_data.items():
+            view_data[view_column_names[view_key]] = view_settings['enabled']
+            if (view_key == 'gd' and 'show_grade_counts' in view_settings.keys()):
+                Course.objects.filter(pk=course_id).update(
+                    show_grade_counts=view_settings['show_grade_counts'])
+
+        CourseViewOption.objects.filter(pk=course_id).update(**view_data)
+    except (ObjectDoesNotExist, Exception) as e:
+        logger.info(
+            f'updating course visualization options failed due to {e} for user {current_user} in course: {course_id}')
```

Comment:
I had to resist the urge to wrap the variables the way I'd like them, as `"{current_user}"` and `({course_id}}`.  I like to do that so I can see in the log messages whether these things are strings or numbers and where are the ends.  Especially because strings may have spaces at either end.